### ISSUE TITLE
Crashes the surefire JVM

### ISSUE THREAD

**Issue by delanym** [junit-pioneer/junit-pioneer#607]
Im using maven-surefire-plugion:3.0.0-M5 and junit 5.8.2

In a project I simply add the dependency and it crashes the JVM.

``` xml
      <dependency>
        <groupId>org.junit-pioneer</groupId>
        <artifactId>junit-pioneer</artifactId>
        <version>1.4.0</version>
      </dependency>
```

I was fine with 1.3.8

```
[ERROR] TestEngine with ID 'junit-jupiter' failed to discover tests
[ERROR] org.apache.maven.surefire.booter.SurefireBooterForkException: There was an error in the forked process
[ERROR] TestEngine with ID 'junit-jupiter' failed to discover tests
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.fork(ForkStarter.java:733)
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:305)
[ERROR]         at org.apache.maven.plugin.surefire.booterclient.ForkStarter.run(ForkStarter.java:265)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeProvider(AbstractSurefireMojo.java:1314)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.executeAfterPreconditionsChecked(AbstractSurefireMojo.java:1159)
[ERROR]         at org.apache.maven.plugin.surefire.AbstractSurefireMojo.execute(AbstractSurefireMojo.java:932)
[ERROR]         at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
[ERROR]         at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
[ERROR]         at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
[ERROR]         at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:196)
[ERROR]         at org.apache.maven.lifecycle.internal.builder.multithreaded.MultiThreadedBuilder$1.call(MultiThreadedBuilder.java:186)
[ERROR]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[ERROR]         at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515)
[ERROR]         at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
[ERROR]         at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
[ERROR]         at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
[ERROR]         at java.base/java.lang.Thread.run(Thread.java:829)
```

**Comment by beatngu13** [comment]
Hi @delanym,

JUnit Pioneer 1.4.0 is an old release from 28 April 2021, that was [depending on JUnit Jupiter 5.7.1](https://github.com/junit-pioneer/junit-pioneer/blob/v1.4.0/gradle.properties#L1). From the ["Dependencies" section of our `README.md`](https://github.com/junit-pioneer/junit-pioneer#dependencies):

>Pioneer does not only use JUnit 5's API, but also other artifacts from its ecosystem such as [`junit-platform-commons`](https://mvnrepository.com/artifact/org.junit.platform/junit-platform-commons).
To avoid dependency issues (e.g. in [junit-pioneer#343](https://github.com/junit-pioneer/junit-pioneer/issues/343)), you should add the JUnit 5 BOM ([`junit-bom`](https://mvnrepository.com/artifact/org.junit/junit-bom)) to your project instead of defining all dependency versions manually.

Therefore, please use the current release 1.6.2, which is compatible with JUnit Jupiter 5.8.2. If you are then still facing this issue, feel free to reopen.

**Comment by delanym** [comment]
I'm saying its been crashing since v1.4.0. Of course I tried with the latest first.

**Comment by delanym** [comment]
I don't have any permissions to reopen this issue

**Comment by Michael1993** [comment]
Please let us know what JUnit version you are using @delanym

**Comment by beatngu13** [comment]
@delanym please provide an [MRE](https://en.wikipedia.org/wiki/Minimal_reproducible_example), ideally a minimal `pom.xml` / Maven project.

And have you tried using the `junit-bom`?

**Comment by delanym** [comment]
It was API incompat - solved using junit-bom.

I came here because a unit test is now failing building under JDK11. The JVM loads locale providers in a different order.
I wanted to switch the test off if the locale was something different from what it was expecting, but the @DefaultLocale annotation sets the locale rather than filtering the test based on the locale.

In the end I just used @DisableOnJre(JRE.JAVA_11)
